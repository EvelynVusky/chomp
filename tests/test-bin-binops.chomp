int main() {
    bit a = {{1}};
    nibble b = {{0101}};
    byte c = {{11100010}};
    word d = {{0111000110011110}};

    /* binand */
    println(a & d);
    println(b & {{10011010}});

    /* binor */
    println(b | c);
    println(d | {{1}});

    /* binxor */
    println(a ^ c);
    println(c ^ {{1011001000011100}});

    return 0;
}