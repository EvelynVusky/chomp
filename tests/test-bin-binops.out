0
0000
0111
1
1
11111110
